### ISSUE TITLE
Can't join #<room>:libera.chat

### ISSUE THREAD

**Issue by freijon** [matrix-org/synapse#15003]
### Description

I'm trying to join #\<room\>:libera.chat but it keeps failing. I keep getting errors on Matrix side like `There was an error joining.` or `Failed to make_join via any server`.

On the IRC side the user exists and joins the channel. If I use `!join channel` I even get an invitation by appservice but when I try to join I get the same error.

My server admin tells me all libera.chat rooms are impacted where no user from our homeserver has joined yet. Sounds like a federation issue?

Please help

### Steps to reproduce

- Try to join #gentoo:libera.chat
- Receive error message

or

- Talk to libera.chat Appservice and use `join #gentoo`
- Receive invitation by appservice
- Join invitation
- After a while, receive error message

### Homeserver

fairydust.space

### Synapse Version

1.76.0

### Installation Method

Docker (matrixdotorg/synapse)

### Database

I don't know

### Workers

I don't know

### Platform

I don't know

### Relevant log output

```shell
I don't have log output, the server admin is quite busy
```

**Comment by vaguerant** [comment]
I'm encountering what appears to be the same issue on my end, on the `matrix.org` homeserver, so you should be able to access logs of this happening that way. I'm `@vaguerant:matrix.org`, attempting (and failing) to join `#godmode9:libera.chat`.

It's a small and niche channel; as in the original filing, I'm likely to be the only user from `matrix.org` (or possibly any Matrix server) attempting to join. I was previously in the room before being idle-kicked and am now unable to rejoin due to this issue.

I'll also add that this element-web issue may be related: vector-im/element-web#22720

**Comment by H-Shay** [comment]
FTR I was able to join #godmode9:libera.chat from matrix.org this morning, what error messages are you seeing when you try to join?

**Comment by vaguerant** [comment]
Opening up Element today, I've been successfully joined to `#godmode9:libera.chat`; perhaps you woke something up by joining? Obviously I can't get the errors to occur any more now, but there were two errors I would get depending on how I tried to join in Element. If I tried to join it as one would normally join a Matrix room (via the room directory/search or the Historical rooms section of Element's room list), I would get an error saying:

`MatrixError: [404] No known servers`

If I tried to join by sending `!join #godmode9` to the Libera bridge bot (`@appservice:libera.chat`), I would receive an invite to join `#godmode9:libera.chat`. Accepting that invite would pinwheel (show a spinning wheel "loading/connecting" animation in the Element interface), then eventually I would receive a fairly non-descriptive error:

`Failed to join`
